service/dynamodb/dynamodbbatchwriter/batchwriter.go

```diff
@@ -0,0 +1,129 @@
+package dynamodbbatchwriter
+
+import (
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/dynamodbiface"
+)
+
+const defaultFlushAmount = 25
+const defaultRequestBufferCap = 50
+
+// BatchWriter wraps a dynamodb client to expose a simple PutItem/DeleteItem
+// API that buffers requests and takes advantage of BatchWriteItem behind
+// the scenes.
+type BatchWriter struct {
+	_ struct{}
+
+	tableName     string
+	client        dynamodbiface.DynamoDBAPI
+	primaryKeys   []string
+	flushAmount   int
+	requestBuffer []dynamodb.WriteRequest
+}
+
+// New creates a new BatchWriter that will write to table `tableName`
+// using `client`.
+//
+// New will return an error if it fails to access the table information with a
+// DescribeTableRequest.
+func New(tableName string, client dynamodbiface.DynamoDBAPI) (*BatchWriter, error) {
+	describeTableReq := client.DescribeTableRequest(&dynamodb.DescribeTableInput{
+		TableName: &tableName,
+	})
+	describeTableOut, err := describeTableReq.Send()
+	if err != nil {
+		return &BatchWriter{}, err
+	}
+	// List of primary keys of the table. We will get them from a
+	// DescribeTable request to DynamoDB.
+	pKeys := []string{}
+	for _, key := range describeTableOut.Table.KeySchema {
+		pKeys = append(pKeys, *key.AttributeName)
+	}
+	requestBuffer := make(
+		[]dynamodb.WriteRequest, 0, defaultRequestBufferCap,
```

Comment:
Currently, they can't. Do you think it would be a worthwhile feature? I've no problem with adding it as a parameter to the New function. In that case, I would prefer to wrap the options in a `BatchWriterOptions` struct because I assume most users would be happy with the default capacity.

That would make it easy to add `FlushAmount` there as well.

Reply:
I think exposing `FlushAmount` should be good and having the cap be passed in as a parameter should be a happy median. For consistency sake, we should follow what other high level abstractions are doing, `s3manager` and `expressions`. They use functional options or parameters.

Reply:
I'll take a better look at `s3manager` to see what else I can get from there.